src/averageAges.js

```diff
@@ -37,7 +42,16 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const women = withChildren
+    ? people.filter((p) => p.sex === 'f'
+      && people.find((p1) => p1.mother === p.name))
```

Comment:
If, for example, we need to find the mother among the array, it is better to use the "some" method,
  which returns a boolean value. That is, we do not need to return the mother, thereby increasing the volume of the return object.
  It is enough for us to know whether there is a mother or not.